Source/WebCore/page/Navigation.h

```diff
@@ -195,6 +197,28 @@ class Navigation final : public RefCounted<Navigation>, public EventTarget, publ
     static Vector<Ref<HistoryItem>> filterHistoryItemsForNavigationAPI(Vector<Ref<HistoryItem>>&&, HistoryItem&);
 
 private:
+    // Rate limiter to prevent excessive navigation requests.
+    class NavigationRateLimiter {
+    public:
+        NavigationRateLimiter() = default;
+        NavigationRateLimiter(const Navigation&) = delete;
```

Comment:
Use `WTF_MAKE_NONCOPYABLE` and `WTF_MAKE_NONMOVABLE` macros.

Reply:
These `Navigation` should be this class name, right? That's one of the reason we use macro.